### PR TITLE
Bump and unify PDM in the api/ and indexer_worker/ to 2.21

### PR DESCRIPTION
**Pull request by krysal**
## Description

Follows the series of dependency updates covering Renovate missings.

## Testing Instructions

Confirm the API continues to run normally. Watch the CI workflow passing.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (`ov just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/Dockerfile [hunk 1]
```diff
@@ -19,11 +19,11 @@ ENV PIP_NO_COLOR=1
...
 # - Install PDM to install Python dependencies
 RUN apt-get update \
-  && apt-get install -yqq --no-install-recommends \
-    python3-dev \
+  && apt-get install -yqq --no-install-recommends python3-dev \
   && apt-get autoremove -y \
   && rm -rf /var/lib/apt/lists/* \
-  && pip install pdm~=2.19
+  && pip install --upgrade pip \
+  && pip install pdm~=2.21
```
Comment (AetherUnbound) [inline api/Dockerfile:26]: We have the Python version defined as `API_PY_VERSION` on Line 5, does it maybe make sense to make this package version an similar argument in the same place in the Dockerfile? Even if it just has a default to the pinned version.

**Review by AetherUnbound — APPROVED** [review]
One note on version placement, but otherwise this works locally for me and looks good!

File: api/Dockerfile (hunk 1 above)
Comment (dhruvkb) [reply api/Dockerfile:26]: That makes sense, it would also make it easier to bump PDM everywhere by changing it in one place!

**Review by dhruvkb — APPROVED** [review]
Approving this, but with a +1 to @AetherUnbound's idea. Up to you if you want to pursue that as a separate PR.

**Comment by krysal** [comment]
Folks, I managed to centralize the version in an env var from the justfile. It was all about the placement since, in the Dockerfile, each `FROM` line starts a new image and resets the build environment. I was placing it in the previous context. Thanks for making the suggestion! This is a much better solution 😄

**Review by dhruvkb — APPROVED** [review]
This is great! 

We will need to manually remember to update PDM from time to time because Renovate would not look for the version number in `justfile`.

**Comment by krysal** [comment]
@dhruvkb Correct, the same situation as currently because renovate wasn't updating PDM anyway.

**Comment by dhruvkb** [comment]
True, that was just a mental note and not even remotely a critique of this PR.

**Comment by AetherUnbound** [comment]
Ahhh, that's excellent! I'm glad it was possible to centralize that!

**Pull request merged**